controller/k8s/api.go

```diff
@@ -330,12 +330,24 @@ func (api *API) GetObjects(namespace, restype, name string) ([]runtime.Object, e
 // singular resource type (e.g. deployment, daemonset, job, etc.)
 func (api *API) GetOwnerKindAndName(pod *corev1.Pod) (string, string) {
 	if len(pod.GetOwnerReferences()) != 1 {
+		log.Debugf("unexpected owner reference count (%d): %+v", len(pod.GetOwnerReferences()), pod.GetOwnerReferences())
```

Comment:
what are the situations where this can arise?  If it's unexpected, it sounds like an error. If this happens in some known case, can we be more descriptive?

Reply:
`len(pod.GetOwnerReferences()) == 0` if it's a pod without a parent. i'm not sure if `> 1` is possible. this check was added in 252a8d39d3dc241e9ec6d6eae11490d5c5e144a5 as part of a larger change. the intent of adding a debug line was to learn if it is possible. i can break this up into two separate checks, but i'd be reluctant to change this into an error without understanding the situations that can cause it.